Dynamic mode options

ShiftyRail opened the pull request:
[system] [administration]

![image](https://user-images.githubusercontent.com/31417754/56856443-b5d32e80-695a-11e9-9f2d-0e00d628a1f7.png)

See picture above.

The most important option is **no stacking**.
If it's active, the following rulesets are mutually exclusive :
- Revolution (any kind)
- Blob (any kind)
- Nuke ops (any kind)
- Cult
- Malf

Fully tested & all

boy2mantwicethefam commented:
Making a really good PR but also delivering a change you want that would need people to vote for. No stacking should be off by default and then toggled on by default in a separate PR mostly to preserve atomicity

ShiftyRail commented:
That's fairly true

boy2mantwicethefam commented on code/modules/admin/admin.dm:
```diff
@@ -732,6 +731,37 @@ var/global/floorIsLava = 0
+		<a href='?src=\ref[src];classic_secret=1'>Classic secret mode</a> <br/>
+		<b>Description:</b> Only one roundstart ruleset will be drafted. Only traitors and minor roles will latespawn. This option is currently <b>[dynamic_classic_secret ? "on" : "off"]</b>. <br/>
+		<a href='?src=\ref[src];high_pop_limit=1'>High population limit</a> <br/>
+		<b>Description:</b>The threshold at which "high pipulation overide" will be in effect. The current value is <b>[dynamic_classic_secret ? "on" : "off"]</b>. <br/>
```
override

SirSkeleto replied:
pipulation

boy2mantwicethefam commented on code/modules/admin/admin.dm:
```diff
@@ -732,6 +731,37 @@ var/global/floorIsLava = 0
+		<center><B><h3>Dynamic Mode Options</h3></B></center><hr>
+		<br/> 
+		<h4>Common options</h4>
+		<i>All thse options can be changed midround.</i> <br/>
```
these

ShiftyRail commented:
Changed the no stacking default option for atomicity & all

SonixApache commented:
I was wondering if such a control panel would ever show up. Really good shit. 👌👌👌👌👌👌

Lokiamis commented:
pipulation

oversane commented:
B-but no stacking means we will basically never get to see the unique interaction between nuke ops and a blob!

ShiftyRail commented:
![image](https://user-images.githubusercontent.com/31417754/56854082-4eed4f80-6931-11e9-981b-bc316a5daa9e.png)

Made the interface a bit clearer, added some links for changing the parameters midround.
Now need to test if the features actually work

ShiftyRail commented:
Added some procs to make testing of mutually exclusive roundstart easier (can comment them out if needed)
Fixed roundstart rulesets not being mutually exclusives with one another
Added a new variable to control (the threshold at which several round enders can happen ; the default is 90, which under normal distribution happens 2% of the rounds)

I spend a good 2 hours testing this so I'd say it's ready for review :tm:

robosperg commented:
dangerously based PR

DamianX commented:
Does this get reset every round? Persistence would be nice.

ShiftyRail commented:
I'm not sure. If a badmin force extended for a bus but forgets to untick it after round end, it could be really frustrating in my opinion.

Once this technical PR is merged, I will make a PR to change the default values (no stacking to 1 by default).

DeityLink commented:
I think having a stacking treshold like you did is a better alternative to outright preventing stacking.

Could you possibly add a forced starting Threat Level?

ShiftyRail commented:
@DeityLink
That is the "rigged threat number" setting.
